.github/workflows/docsbuild-release.yaml

```diff
@@ -0,0 +1,73 @@
+name: ansible-release-porting-guide
+on:
+  workflow_dispatch:
+    inputs:
+      ansible_version:
+        description: 'Release Version. Example : 11.1.0'
+        required: true
+      ansible_major_version:
+        description: 'Exmaple 11'
+        required: true
+
+env:
+  CI_COMMIT_MESSAGE: 'Add Ansible community "${{ inputs.ansible_version }}" porting guide'
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  CI_PR_BODY: |
+
+      ##### SUMMARY
+
+      ##### ISSUE TYPE
+
+      -   Docs Pull Request
+
+      ##### COMPONENT NAME
+
+      docs/docsite/rst/porting_guides/'porting_guide_"${{ inputs.ansible_major_version }}".rst'
+
+jobs:
+  build:
+    name: 'Build Ansible community distribution (${{ inputs.ansible_version }})'
+    runs-on: ubuntu-latest
+
+    steps:
+
+# checkout the /ansible/ansible-documentation & ansible-build-data
+      - name: Check out ansible-documentation
+        uses: actions/checkout@v3
+        with:
+          path: ansible-documentation
+          ref: devel
+
+      - name: Checking out to a new branch & setting the user details
+        run: |
+          pwd
+          ls
+          cd ansible-documentation
+          git checkout -b "publish-${{ inputs.ansible_version }}"
+          git config --global user.name "${{github.actor}}"
+          git config --global user.email "${{github.actor}}@users.noreply.github.com"
```

Comment:
This is an old-style email, it may become obsolete and point to somebody else over time. The correct solution is to use an ID in it.
https://hynek.me/til/easier-crediting-contributors-github/

Also, here's an action example that allows making GHA the author: https://github.com/fregante/setup-git-user.